### DISCUSSION TITLE
Access content from createData without creating a new component

### DISCUSSION THREAD

**Discussion by atomicpages** [facebook/docusaurus#4623] [Q&A]
Hey there 👋 ,

is there a way to access content created via `createData` without using `addRoute`? Something like:

```js
const { data } = usePluginData('something-rad');
```

but where `someting-rad` is created like so:

```js
export default function plugin() {
  return {
    name: 'someting-rad',
    async loadContent() { ... },
    async contentLoaded({ content, actions }) {
      createData('woot.json', JSON.stringify(content));
    }
}
```

**Comment by slorber** [comment]
What do you mean by "access content", where do you want to access this data exactly and what does it contain?

createData/addRoute is mostly useful when you want a route component to receive the data as props

You can import/require any json file anywhere in your React code, so you can just write `woot.json` anywhere (in a plugin hook or a prebuild step) and import it with `import Woot from "@site/woot.json"` for example.

There is also a "global data"  API that you can use to create some data that will be available everywhere by default
https://docusaurus.io/docs/next/docusaurus-core#useglobaldata

Keep  in mind that you should avoid using such API if the data is useful in only a subset of pages, because it will increase the weight of all your pages of your site (including the homepage)

**Reply by atomicpages** [reply]
Ah, gotcha. Specifically this is for [docusaurus-plugin-react-docgen-typescript](https://github.com/atomicpages/docusaurus-plugin-react-docgen-typescript). From `loadContent` I'm using `createData` to load type annotations into a json file within `.docusaurus`. 

I did some digging with aliases and found `@generated` which does what I need. Didn't find any documentation on it, however.

**Reply by slorber** [reply]
We don't provide doc for `@generated` because it's not really worth documenting, it's just a shortcut for docusaurus to reduce length of paths and bundle size, but really we could just write `@site/.docusaurus` instead, (and `@site` is documented.

You can also require any json file with relative path or set your own webpack aliases. Requirig a file from an alias, an absolute path or relative path can work in webpack and Docusaurus support all that.
